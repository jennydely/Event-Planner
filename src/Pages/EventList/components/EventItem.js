import React, { useState, useContext } from 'react'
import { useSpring } from 'react-spring'
import styled from 'styled-components/macro'
import packlistIcon from '../../../img/packlistIcon.svg'
import editIcon from '../../../img/editIcon.svg'
import starIcon from '../../../img/starIcon.svg'
import starredIcon from '../../../img/starredIcon.svg'
import copyIcon from '../../../img/copyIcon.svg'
import getColorOfEventCategory from '../../../services/getColorOfEventCategory'
import { formatDate } from '../../../services/date'
import EventDetails from './EventDetails'
import useHeight from './useHeight'
import { useHistory } from 'react-router-dom'
import { UserContext } from "../../../providers/UserProvider";
import { putEvent } from '../services/putEvent'
import { useDispatch } from 'react-redux'

export default function EventItem({
  event,
  id,
  filteredEventList,
  onTicketCheckboxClick,
}) {
  const {
    name,
    location,
    category,
    eventStartDate,
    eventEndDate,
    packlistCategory,
  } = event
  const { height, bind } = useHeight([event])
  const [isEventDetailVisible, setIsEventDetailVisible] = useState(false)
  const [isStarred, setIsStarred] = useState(event.isStarred)
  const detailStyle = {
    ...useSpring({
      height: isEventDetailVisible ? height : 0,
    }),
  }
  const history = useHistory()
  const user = useContext(UserContext);
  const dispatch = useDispatch()

  function handleEditEvent(eventId) {
    history.push('/eventform/edit/' + eventId)
  }

  function handleCopyEvent(eventId) {
    history.push('/eventform/copy/' + eventId)
  }

  function showPacklist() {
    history.push('/packlist/' + packlistCategory)
  }

  return (
    <>
      <Event>
        <EventHeader name={category} onClick={toggleEventDetail}>
          <TextContainer>
            <h2>~ {name} ~</h2>
            <h3>{location}</h3>
            <h3>{formatDate(eventStartDate, eventEndDate)}</h3>
          </TextContainer>

          {user && filteredEventList.find((filteredEvent) => filteredEvent.visibility === 'private') ?
            < ButtonContainer >
              <PacklistButton onClick={showPacklist}>
                <img src={packlistIcon} alt="packlist" />
              </PacklistButton>
              <EditButton onClick={handleEditButtonClick}>
                <img src={editIcon} alt="edit" />
              </EditButton>
            </ButtonContainer>
            :
            user && filteredEventList.find((filteredEvent) => filteredEvent.visibility === 'public') ?
              < ButtonContainer >
                <PacklistButton onClick={toggleStar}>
                  {event.isStarred ? <img src={starredIcon} alt="starred" /> : <img src={starIcon} alt="star" />}
                </PacklistButton>
                <EditButton onClick={handleCopyButtonClick}>
                  <img src={copyIcon} alt="copy" />
                </EditButton>
              </ButtonContainer>
              :
              user && filteredEventList.find((filteredEvent) => filteredEvent.isStarred === true) &&
              < ButtonContainer >
                <PacklistButton onClick={toggleStar}>
                  {event.isStarred ? <img src={starredIcon} alt="starred" /> : <img src={starIcon} alt="star" />}
                </PacklistButton>
                <EditButton onClick={handleEditButtonClick}>
                  <img src={editIcon} alt="edit" />
                </EditButton>
              </ButtonContainer>
          }

        </EventHeader>
        {user ?
          <EventDetails
            event={event}
            style={detailStyle}
            bind={bind}
            id={id}
            onTicketCheckboxClick={onTicketCheckboxClick}
          />
          :
          <EventDetails
            event={event}
            style={detailStyle}
            bind={bind}
            id={id}
          />}
      </Event>
    </>
  )

  function toggleEventDetail() {
    setIsEventDetailVisible(!isEventDetailVisible)
  }

  function toggleStar(ev) {
    ev.preventDefault()
    ev.stopPropagation()
    setIsStarred(!isStarred)
    event.isStarred = !isStarred
    dispatch({
      type: "UPDATE_EVENT", payload: {
        ...event
      }
    })
  }

  function handleEditButtonClick() {
    handleEditEvent(event.id)
  }

  function handleCopyButtonClick() {
    handleCopyEvent(event.id)
  }

}


const Event = styled.li`
  list-style: none;
  margin: 0;
  margin-bottom: 20px;
  padding: 4px;
  text-align: center;
`
const EventHeader = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 0;
  padding: 4px 0 4px 4px;
  border-radius: 7px;
  background: var(--eventItem);
  border: solid 3px ${(props) => getColorOfEventCategory(props.name)};
  border-left: solid 10px ${(props) => getColorOfEventCategory(props.name)};
`
const TextContainer = styled.div`
  align-self: center;
  flex-basis: 90%;
  max-width: 365px;
`
const ButtonContainer = styled.div`
  align-items: flex-end;
  flex-basis: 10%;
`
const PacklistButton = styled.button`
  padding: 22px 11px 2px 0;
`
const EditButton = styled.button`
  padding: 2px 11px 22px 0;
`
