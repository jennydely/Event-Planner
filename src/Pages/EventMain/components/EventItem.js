import React, { useState } from 'react'
import { useSpring } from 'react-spring'
import styled from 'styled-components/macro'
import eyeIcon from '../../../img/eyeIcon.svg'
import hideEyeIcon from '../../../img/hideEyeIcon.svg'
import editIcon from '../../../img/editIcon.svg'
import getColorOfEventCategory from '../../../services/getColorOfEventCategory'
import { formatDate } from '../../../services/date'
import EventDetails from './EventDetails'
import useHeight from './useHeight'
import { useHistory } from 'react-router-dom'

export default function EventItem({
  event,
  id,
  onHideButtonClick,
  onDeleteButtonClick,
  onTicketCheckboxClick,
}) {
  const { name, location, category, eventStartDate, eventEndDate } = event
  const { height, bind } = useHeight([event])
  const [isEventDetailVisible, setIsEventDetailVisible] = useState(false)
  const detailStyle = {
    ...useSpring({
      height: isEventDetailVisible ? height : 0,
    }),
  }
  const history = useHistory()
  function handleEditEvent(eventId) {
    history.push('/eventform/' + eventId)
  }

  return (
    <>
      <Event>
        <EventHeader name={category} onClick={toggleEventDetail}>
          <TextContainer>
            <h2>~ {name} ~</h2>
            <h3>{location}</h3>
            <h3>{formatDate(eventStartDate, eventEndDate)}</h3>
          </TextContainer>
          <ButtonContainer>
            <HideButton onClick={handleHideButtonClick} id={id}>
              {event.isHidden ? (
                <img src={eyeIcon} alt="show" />
              ) : (
                <img src={hideEyeIcon} alt="hide" />
              )}
            </HideButton>
            <EditButton onClick={handleEditButtonClick}>
              <img src={editIcon} alt="edit" />
            </EditButton>
          </ButtonContainer>
        </EventHeader>
        <EventDetails
          event={event}
          style={detailStyle}
          bind={bind}
          onDeleteButtonClick={onDeleteButtonClick}
          onTicketCheckboxClick={onTicketCheckboxClick}
        />
      </Event>
    </>
  )

  function toggleEventDetail() {
    setIsEventDetailVisible(!isEventDetailVisible)
  }

  function handleHideButtonClick() {
    onHideButtonClick(event.id)
  }

  function handleEditButtonClick() {
    handleEditEvent(event.id)
  }
}

const Event = styled.li`
  list-style: none;
  margin: 0;
  margin-bottom: 20px;
  padding: 4px;
  text-align: center;
`
const EventHeader = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 0;
  padding: 4px 0 4px 4px;
  border-radius: 6px;
  border: solid 3px ${(props) => getColorOfEventCategory(props.name)};
  border-left: solid 10px ${(props) => getColorOfEventCategory(props.name)};
`
const TextContainer = styled.div`
  align-self: center;
  flex-basis: 98%;
`
const ButtonContainer = styled.div`
  align-items: flex-end;
`
const HideButton = styled.button`
  padding: 22px 0 2px 11px;
`
const EditButton = styled.button`
  padding: 2px 0 22px 11px;
`
