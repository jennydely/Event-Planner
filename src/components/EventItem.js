import React, { useState } from 'react'
import { useSpring } from 'react-spring'
import useHeight from './useHeight'
import styled from 'styled-components/macro'
import { formatDate } from '../services/date'
import EventDetails from './EventDetails'
import eyeIcon from '../img/eyeIcon.svg'
import hideEyeIcon from '../img/hideEyeIcon.svg'
import getColorByName from '../services/getColorByName'

export default function EventItem({
  event,
  id,
  onHideButtonClick,
  onDeleteButtonClick,
  onTicketCheckboxClick,
}) {
  const { name, location, category, eventStartDate, eventEndDate } = event
  const { height, bind } = useHeight([event])
  const [isEventDetailVisible, setIsEventDetailVisible] = useState(false)
  const detailStyle = {
    ...useSpring({
      height: isEventDetailVisible ? height : 0,
    }),
  }

  return (
    <>
      <Event>
        <EventHeader name={category} onClick={toggleEventDetail}>
          <h2>~ {name} ~</h2>
          <h3>{location}</h3>
          <h3>{formatDate(eventStartDate, eventEndDate)}</h3>
        </EventHeader>
        <EventDetails
          event={event}
          style={detailStyle}
          bind={bind}
          onDeleteButtonClick={onDeleteButtonClick}
          onTicketCheckboxClick={onTicketCheckboxClick}
        />
      </Event>
      <HideButton onClick={handleHideButtonClick} id={id}>
        {event.isHidden ? (
          <img src={eyeIcon} alt="show" />
        ) : (
          <img src={hideEyeIcon} alt="hide" />
        )}
      </HideButton>
    </>
  )

  function toggleEventDetail() {
    setIsEventDetailVisible(!isEventDetailVisible)
  }

  function handleHideButtonClick() {
    onHideButtonClick(event.id)
  }
}

const Event = styled.li`
  grid-column: 1;
  list-style: none;
  margin: 0;
  margin-bottom: 20px;
  padding: 4px;
  text-align: center;
`
const EventHeader = styled.div`
  margin: 0;
  border-radius: 6px;
  border: solid 3px ${(props) => getColorByName(props.name)};
  border-left: solid 10px ${(props) => getColorByName(props.name)};
`
const HideButton = styled.button`
  grid-column: 2;
  align-self: start;
  justify-self: start;
  margin: 0;
  padding: 5px 0;
  margin-top: 23.325px;
  background: none;
`
