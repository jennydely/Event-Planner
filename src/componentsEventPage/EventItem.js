import React from 'react';
import styled from 'styled-components/macro'
import {formatDate} from '../utils/date'
import EventDetails from './EventDetails'

export default function EventItem({event}) {
const{name, location, eventStartDate, eventEndDate} = event

    return (
        <Event>
            <Title>{name} - {location}</Title>
            <Duration>{formatDate(eventStartDate, eventEndDate)}</Duration>
            <EventDetails Hidden event={event} />
        </Event>
    )
}



const Event = styled.li`
list-style:none;
border: 2px solid #964B00;
marign: 0;
margin-bottom:20px;
padding:4px;
text-align:center;
`
const Title = styled.h2`
margin: 0;
font-size: 100%;
`
const Duration = styled.p`
margin:0;
`