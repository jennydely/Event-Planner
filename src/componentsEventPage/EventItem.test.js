import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import EventItem from './EventItem'
import 'jest-styled-components'

describe('EventItem', () => {
    const name = 'This is the eventname'
    const location = 'This is the eventlocation'
    const inputStartDate = '2020-06-25'
    const inputEndDate = '2020-06-28'
    const outputStartDate = '06/25/2020'
    const outputEndDate = '06/28/2020'


    it('displays the eventname and eventlocation', () => {
        const { getByText } = render(<EventItem event={{ name, location, startDate: inputStartDate, endDate: inputEndDate }} />)
        setTimeout(() => {
            expect(getByText(name + ' - ' + location)).toBeInTheDocument()
            expect(getByText(outputStartDate + '-' + outputEndDate)).toBeInTheDocument()
        }, 100)
    })
    it('renders correctly', () => {
        const tree = renderer.create(<EventItem event={{ name, location, startDate: inputStartDate, endDate: inputEndDate }} />)
        expect(tree).toMatchSnapshot()
    })
})