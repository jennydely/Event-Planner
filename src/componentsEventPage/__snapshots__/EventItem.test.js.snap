// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventItem renders correctly 1`] = `
.c2 {
  display: grid;
  grid-template-columns: 20% repeat(4,auto);
  grid-template-rows: repeat(7,auto) 30px auto;
  margin: 0;
  margin-top: -2px;
  border: 2px solid rgb(96,99,104);
  border-top: 0;
  overflow: hidden;
}

.c2::before,
.c2::after {
  content: '';
  position: absolute;
  left: 0;
  width: 100%;
}

.c2::before {
  height: 0;
}

.c2::after {
  height: 8px;
  bottom: 0;
}

.c3 {
  grid-column: 1;
  grid-row: 1 / span 5;
}

.c4 {
  grid-column: 1 /span 1;
  grid-row: 1 / span 5;
  margin: 7px;
  object-fit: cover;
  width: 90%;
  max-height: 90%;
}

.c5 {
  grid-column: 2;
  grid-row: 1/ span 4;
  text-align: right;
  margin: 4px;
  margin-top: 7px;
  font-weight: bold;
  font-size: 100%;
}

.c6 {
  grid-column: 2;
  grid-row: 5;
  text-align: right;
  margin: 4px;
  font-weight: bold;
  font-size: 100%;
}

.c7 {
  grid-column: 3/5;
  grid-row: 1;
  text-align: left;
  margin: 4px;
  margin-top: 7px;
  font-size: 100%;
}

.c8 {
  grid-column: 3/5;
  grid-row: 2;
  text-align: left;
  margin: 4px;
  font-size: 100%;
}

.c9 {
  grid-column: 3/5;
  grid-row: 3;
  text-align: left;
  margin: 4px;
  font-size: 100%;
}

.c10 {
  grid-column: 3/5;
  grid-row: 5;
  text-align: left;
  margin: 4px;
  font-size: 100%;
}

.c11 {
  type: checkbox;
  display: none;
}

.c12 {
  grid-column: 1/6;
  grid-row: 6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  with: 100%;
  padding: 7px;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c13 {
  display: inline-block;
  with: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  margin: 2px;
  padding: 0.3px 4px;
  border: solid 2px rgb(49,42,42);
  border-radius: 6px;
  background-color: rgba(49,42,42,0.75);
  color: rgb(187,148,87);
  font-size: 100%;
  text-align: center;
}

.c0 {
  list-style: none;
  marign: 0;
  margin-bottom: 20px;
  padding: 4px;
  text-align: center;
}

.c1 {
  margin: 0;
  background: rgb(96,99,104);
}

<li
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <h2>
      This is the eventname
       - 
      This is the eventlocation
    </h2>
    <h3>
      Invalid Date - Invalid Date
    </h3>
  </div>
  <section
    className="c2"
    style={
      Object {
        "height": 0,
      }
    }
  >
    <a
      className="c3"
      target="blank"
    >
       
      <img
        alt="This is the eventname Poster"
        className="c4"
        src="https://raw.githubusercontent.com/jennydely/Event-Planner/EventDetails/src/img/defaultImg.jpg"
      />
    </a>
    <p
      className="c5"
    >
      Address: 
    </p>
    <p
      className="c6"
    >
      Price: 
    </p>
    <p
      className="c7"
    >
      This is the eventname
    </p>
    <p
      className="c8"
    />
    <p
      className="c9"
    >
       
      This is the eventlocation
    </p>
    <p
      className="c10"
    >
      kostenlos
       
    </p>
    <button
      className="c11"
    />
    <div
      className="c12"
    >
      <button>
        PackList
      </button>
      <a
        className="c13"
        target="blank"
        title="link"
      >
        Website
      </a>
      <a
        className="c13"
        href="https://www.google.de/maps"
        target="blank"
        title="link"
      >
        Googlemaps
      </a>
    </div>
  </section>
</li>
`;
